src/IsotonicFPOP.cpp

```diff
@@ -40,18 +41,39 @@ void IsotonicFPOP
       cost->piece_list.emplace_back
       (1.0, -2 * data_vec[0], data_vec[0] * data_vec[0],
        min_mean, max_mean, -1, false);
-      // cost -> print();
     }else{ // Alg 3 ln 6 - 8 
-      min_prev_cost.set_to_min_less_of(cost_prev, verbose);
+      
+      if (*constraint) {
+        min_prev_cost.set_to_min_less_of(cost_prev, verbose);  
+      } else {
+        min_prev_cost.set_to_unconstrained_min_of(cost_prev, verbose);  
+      }
+
       min_prev_cost.set_prev_seg_end(data_i-1);
       min_prev_cost.add(0.0, 0.0, penalty);
       cost->set_to_min_env_of(&min_prev_cost, cost_prev, verbose);
+      
+      if (verbose) {
+      printf("new cost - likelihood \n");
+      cost -> print();
+      }
+      
       cost->add
         (1.0,
-         -2 * data_vec[data_i], data_vec[0] * data_vec[0]);
+         -2 * data_vec[data_i], data_vec[data_i] * data_vec[data_i]);
+      
+      // printf("new cost*\n");
+      // cost -> print();
     }
     cost_prev = cost;
   }
+
+  if (verbose) {
+  // Print mu functions 
+  printf("Printing Cost*_T(mu) functions to get handle on tau*_T(mu) \n");
+  cost -> print();
```

Comment:
Cool, thanks! I guess I need to make some small adjustments for the square loss, but that should be straightforward (I think!).